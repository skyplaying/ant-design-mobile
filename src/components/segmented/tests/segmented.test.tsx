import React, { useState } from 'react'
import { render, testA11y, fireEvent } from 'testing'
import { AppOutline, UnorderedListOutline } from 'antd-mobile-icons'

import Segmented from '..'
import type { SegmentedValue } from '..'

const classPrefix = `adm-segmented`

function expectMatchChecked(container: HTMLElement, checkedList: boolean[]) {
  const inputList = Array.from(
    container.querySelectorAll<HTMLInputElement>(`.${classPrefix}-item-input`)
  )
  expect(inputList).toHaveLength(checkedList.length)

  inputList.forEach((input, i) => {
    const checked = checkedList[i]

    expect(input.checked).toBe(checked)
  })
}

describe('adm-segmented', () => {
  test('a11y', async () => {
    await testA11y(
      <Segmented
        options={['Daily', 'Weekly', 'Monthly', 'Quarterly', 'Yearly']}
      />
    )
  })

  test('render empty segmented', () => {
    const { asFragment } = render(<Segmented options={[]} />)
    expect(asFragment().firstChild).toMatchSnapshot()
  })

  test('render segmented ok', () => {
    const { asFragment, container } = render(
      <Segmented
        options={[{ label: 'Daily', value: 'Daily' }, 'Weekly', 'Monthly']}
      />
    )

    expect(asFragment().firstChild).toMatchSnapshot()

    expectMatchChecked(container, [true, false, false])
  })

  test('render label with ReactNode', () => {
    const { asFragment, container } = render(
      <Segmented
        options={[
          { label: 'Daily', value: 'Daily' },
          { label: <div id='weekly'>Weekly</div>, value: 'Weekly' },
          { label: <div className='little'>Monthly</div>, value: 'Monthly' },
        ]}
      />
    )

    expect(asFragment().firstChild).toMatchSnapshot()

    expectMatchChecked(container, [true, false, false])

    expect(container.querySelector('#weekly')?.textContent).toContain('Weekly')
    expect(container.querySelector('.little')?.textContent).toContain('Monthly')
  })

  test('render segmented with defaultValue', () => {
    const { container } = render(
      <Segmented
        options={['Daily', 'Weekly', 'Monthly', 'Quarterly', 'Yearly']}
        defaultValue='Quarterly'
      />
    )

    expectMatchChecked(container, [false, false, false, true, false])
  })

  test('render segmented with string options', () => {
    const handleValueChange = jest.fn()
    const { asFragment, container } = render(
      <Segmented
        options={['Daily', 'Weekly', 'Monthly']}
        onChange={handleValueChange}
      />
    )
    expect(asFragment().firstChild).toMatchSnapshot()

    expectMatchChecked(container, [true, false, false])
    expect(
      container
        .querySelectorAll(`label.${classPrefix}-item`)[0]
        .classList.contains(`${classPrefix}-item-selected`)
    ).toBeTruthy()

    fireEvent.click(container.querySelectorAll(`.${classPrefix}-item-input`)[2])
    expect(handleValueChange).toHaveBeenCalledWith('Monthly')

    expectMatchChecked(container, [false, false, true])
  })

  test('render segmented with numeric options', () => {
    const handleValueChange = jest.fn()
    const { asFragment, container } = render(
      <Segmented
        options={[1, 2, 3, 4, 5]}
        onChange={value => handleValueChange(value)}
      />
    )
    expect(asFragment().firstChild).toMatchSnapshot()
    expectMatchChecked(container, [true, false, false, false, false])

    fireEvent.click(container.querySelectorAll(`.${classPrefix}-item-input`)[4])
    expect(handleValueChange).toHaveBeenCalledWith(5)

    expectMatchChecked(container, [false, false, false, false, true])
  })

  test('render segmented with mixed options', () => {
    const handleValueChange = jest.fn()
    const { asFragment, container } = render(
      <Segmented
        options={['Daily', { label: 'Weekly', value: 'Weekly' }, 'Monthly']}
        onChange={value => handleValueChange(value)}
      />
    )
    expect(asFragment().firstChild).toMatchSnapshot()
    expectMatchChecked(container, [true, false, false])

    fireEvent.click(container.querySelectorAll(`.${classPrefix}-item-input`)[1])
    expect(handleValueChange).toHaveBeenCalledWith('Weekly')

    expectMatchChecked(container, [false, true, false])
  })

  test('render segmented with options: disabled', () => {
    const handleValueChange = jest.fn()
    const { asFragment, container } = render(
      <Segmented
        options={[
          'Daily',
          { label: 'Weekly', value: 'Weekly', disabled: true },
          'Monthly',
        ]}
        onChange={value => handleValueChange(value)}
      />
    )
    expect(asFragment().firstChild).toMatchSnapshot()
    expect(
      container
        .querySelectorAll(`label.${classPrefix}-item`)[1]
        .classList.contains(`${classPrefix}-item-disabled`)
    ).toBeTruthy()
    expect(
      container.querySelectorAll(`.${classPrefix}-item-input`)[1]
    ).toHaveAttribute('disabled')

    fireEvent.click(container.querySelectorAll(`.${classPrefix}-item-input`)[1])
    expect(handleValueChange).not.toHaveBeenCalled()

    expectMatchChecked(container, [true, false, false])

    fireEvent.click(container.querySelectorAll(`.${classPrefix}-item-input`)[2])
    expect(handleValueChange).toHaveBeenCalledWith('Monthly')
    expect(handleValueChange).toHaveBeenCalledTimes(1)

    expectMatchChecked(container, [false, false, true])
  })

  test('render segmented: disabled', () => {
    const handleValueChange = jest.fn()
    const { asFragment, container } = render(
      <Segmented
        disabled
        options={['Daily', 'Weekly', 'Monthly']}
        onChange={value => handleValueChange(value)}
      />
    )
    expect(asFragment().firstChild).toMatchSnapshot()
    expect(
      container
        .querySelectorAll(`.${classPrefix}`)[0]
        .classList.contains(`${classPrefix}-disabled`)
    ).toBeTruthy()

    fireEvent.click(container.querySelectorAll(`.${classPrefix}-item-input`)[1])
    expect(handleValueChange).not.toHaveBeenCalled()

    expectMatchChecked(container, [true, false, false])

    fireEvent.click(container.querySelectorAll(`.${classPrefix}-item-input`)[2])
    expect(handleValueChange).not.toHaveBeenCalled()

    expectMatchChecked(container, [true, false, false])
  })

  test('render segmented with className and other html attributes', () => {
    const { container } = render(
      <Segmented
        options={['Daily', 'Monthly', 'Weekly']}
        defaultValue='Weekly'
        className='mock-cls'
        data-test-id='hello'
      />
    )

    expect(container.querySelector('.mock-cls')).toBeTruthy()
    expect(container.querySelector('[data-test-id]')).toHaveAttribute(
      'data-test-id',
      'hello'
    )
  })

  test('render segmented with ref', () => {
    const ref = React.createRef<HTMLDivElement>()
    const { container } = render(
      <Segmented
        options={['Daily', 'Monthly', 'Weekly']}
        defaultValue='Weekly'
        ref={ref}
      />
    )

    expect(ref.current).toBe(container.querySelector(`.${classPrefix}`))
  })

  test('render segmented with controlled mode', async () => {
    const Demo: React.FC = () => {
      const [value, setValue] = useState<SegmentedValue>('Map')
      return (
        <>
          <Segmented
            options={['Map', 'Transit', 'Satellite']}
            value={value}
            onChange={setValue}
          />
          <div className='value'>{value}</div>
          <input
            className='control'
            onChange={e => {
              setValue(e.target.value)
            }}
          />
        </>
      )
    }
    const { container } = render(<Demo />)
    fireEvent.click(container.querySelectorAll(`.${classPrefix}-item-input`)[0])
    expect(container.querySelector('.value')?.textContent).toBe('Map')

    fireEvent.click(container.querySelectorAll(`.${classPrefix}-item-input`)[1])
    expect(container.querySelector('.value')?.textContent).toBe('Transit')
  })

  test('render segmented with options null/undefined', () => {
    const handleValueChange = jest.fn()
    const { asFragment, container } = render(
      <Segmented
        options={[null, undefined, ''] as any}
        disabled
        onChange={value => handleValueChange(value)}
      />
    )
    expect(asFragment().firstChild).toMatchSnapshot()
    expect(
      Array.from(container.querySelectorAll(`.${classPrefix}-item-label`)).map(
        n => n.textContent
      )
    ).toEqual(['', '', ''])
  })

  test('render segmented with thumb', () => {
    const handleValueChange = jest.fn()
    const { asFragment, container } = render(
      <Segmented
        options={['Map', 'Transit', 'Satellite']}
        onChange={value => handleValueChange(value)}
      />
    )
    expect(asFragment().firstChild).toMatchSnapshot()

    expectMatchChecked(container, [true, false, false])
    expect(
      container
        .querySelectorAll(`label.${classPrefix}-item`)[0]
        .classList.contains(`${classPrefix}-item-selected`)
    ).toBeTruthy()

    fireEvent.click(container.querySelectorAll(`.${classPrefix}-item-input`)[2])
    expect(handleValueChange).toHaveBeenCalledWith('Satellite')

    expectMatchChecked(container, [false, false, true])

    // change selection again
    fireEvent.click(container.querySelectorAll(`.${classPrefix}-item-input`)[1])
    expect(handleValueChange).toHaveBeenCalledWith('Transit')

    expectMatchChecked(container, [false, true, false])
  })

  test('render segmented with `block`', () => {
    const { asFragment, container } = render(
      <Segmented block options={['Daily', 'Weekly', 'Monthly']} />
    )

    expect(asFragment().firstChild).toMatchSnapshot()

    expect(
      container
        .querySelectorAll(`.${classPrefix}`)[0]
        .classList.contains(`${classPrefix}-block`)
    ).toBeTruthy()
  })

  test('render with icons', () => {
    const { asFragment, container } = render(
      <Segmented
        options={[
          {
            value: 'List',
            icon: <UnorderedListOutline />,
          },
          {
            value: 'Kanban',
            label: 'KanbanYes',
            icon: <AppOutline />,
          },
        ]}
      />
    )
    expect(asFragment().firstChild).toMatchSnapshot()
    expect(
      container.querySelectorAll(`span.${classPrefix}-item-icon`).length
    ).toBe(2)
    expect(
      container
        .querySelectorAll(`div.${classPrefix}-item-label`)[1]
        .textContent?.includes('KanbanYes')
    ).toBeTruthy()
  })
})
