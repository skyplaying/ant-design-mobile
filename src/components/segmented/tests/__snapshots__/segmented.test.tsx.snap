// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adm-segmented render empty segmented 1`] = `
<div
  aria-label="segmented control"
  class="adm-segmented"
  role="listbox"
>
  <div
    class="adm-segmented-group"
  />
</div>
`;

exports[`adm-segmented render label with ReactNode 1`] = `
<div
  aria-label="segmented control"
  class="adm-segmented"
  role="listbox"
>
  <div
    class="adm-segmented-group"
  >
    <label
      class="adm-segmented-item adm-segmented-item-selected"
    >
      <input
        checked=""
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="true"
        class="adm-segmented-item-label"
        role="option"
        title="Daily"
      >
        Daily
      </div>
    </label>
    <label
      class="adm-segmented-item"
    >
      <input
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
      >
        <div
          id="weekly"
        >
          Weekly
        </div>
      </div>
    </label>
    <label
      class="adm-segmented-item"
    >
      <input
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
      >
        <div
          class="little"
        >
          Monthly
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`adm-segmented render segmented ok 1`] = `
<div
  aria-label="segmented control"
  class="adm-segmented"
  role="listbox"
>
  <div
    class="adm-segmented-group"
  >
    <label
      class="adm-segmented-item adm-segmented-item-selected"
    >
      <input
        checked=""
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="true"
        class="adm-segmented-item-label"
        role="option"
        title="Daily"
      >
        Daily
      </div>
    </label>
    <label
      class="adm-segmented-item"
    >
      <input
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title="Weekly"
      >
        Weekly
      </div>
    </label>
    <label
      class="adm-segmented-item"
    >
      <input
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title="Monthly"
      >
        Monthly
      </div>
    </label>
  </div>
</div>
`;

exports[`adm-segmented render segmented with \`block\` 1`] = `
<div
  aria-label="segmented control"
  class="adm-segmented adm-segmented-block"
  role="listbox"
>
  <div
    class="adm-segmented-group"
  >
    <label
      class="adm-segmented-item adm-segmented-item-selected"
    >
      <input
        checked=""
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="true"
        class="adm-segmented-item-label"
        role="option"
        title="Daily"
      >
        Daily
      </div>
    </label>
    <label
      class="adm-segmented-item"
    >
      <input
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title="Weekly"
      >
        Weekly
      </div>
    </label>
    <label
      class="adm-segmented-item"
    >
      <input
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title="Monthly"
      >
        Monthly
      </div>
    </label>
  </div>
</div>
`;

exports[`adm-segmented render segmented with mixed options 1`] = `
<div
  aria-label="segmented control"
  class="adm-segmented"
  role="listbox"
>
  <div
    class="adm-segmented-group"
  >
    <label
      class="adm-segmented-item adm-segmented-item-selected"
    >
      <input
        checked=""
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="true"
        class="adm-segmented-item-label"
        role="option"
        title="Daily"
      >
        Daily
      </div>
    </label>
    <label
      class="adm-segmented-item"
    >
      <input
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title="Weekly"
      >
        Weekly
      </div>
    </label>
    <label
      class="adm-segmented-item"
    >
      <input
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title="Monthly"
      >
        Monthly
      </div>
    </label>
  </div>
</div>
`;

exports[`adm-segmented render segmented with numeric options 1`] = `
<div
  aria-label="segmented control"
  class="adm-segmented"
  role="listbox"
>
  <div
    class="adm-segmented-group"
  >
    <label
      class="adm-segmented-item adm-segmented-item-selected"
    >
      <input
        checked=""
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="true"
        class="adm-segmented-item-label"
        role="option"
        title="1"
      >
        1
      </div>
    </label>
    <label
      class="adm-segmented-item"
    >
      <input
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title="2"
      >
        2
      </div>
    </label>
    <label
      class="adm-segmented-item"
    >
      <input
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title="3"
      >
        3
      </div>
    </label>
    <label
      class="adm-segmented-item"
    >
      <input
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title="4"
      >
        4
      </div>
    </label>
    <label
      class="adm-segmented-item"
    >
      <input
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title="5"
      >
        5
      </div>
    </label>
  </div>
</div>
`;

exports[`adm-segmented render segmented with options null/undefined 1`] = `
<div
  aria-label="segmented control"
  class="adm-segmented adm-segmented-disabled"
  role="listbox"
>
  <div
    class="adm-segmented-group"
  >
    <label
      class="adm-segmented-item adm-segmented-item-selected adm-segmented-item-disabled"
    >
      <input
        checked=""
        class="adm-segmented-item-input"
        disabled=""
        type="radio"
      />
      <div
        aria-selected="true"
        class="adm-segmented-item-label"
        role="option"
      />
    </label>
    <label
      class="adm-segmented-item adm-segmented-item-disabled"
    >
      <input
        class="adm-segmented-item-input"
        disabled=""
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
      />
    </label>
    <label
      class="adm-segmented-item adm-segmented-item-disabled"
    >
      <input
        class="adm-segmented-item-input"
        disabled=""
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title=""
      />
    </label>
  </div>
</div>
`;

exports[`adm-segmented render segmented with options: disabled 1`] = `
<div
  aria-label="segmented control"
  class="adm-segmented"
  role="listbox"
>
  <div
    class="adm-segmented-group"
  >
    <label
      class="adm-segmented-item adm-segmented-item-selected"
    >
      <input
        checked=""
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="true"
        class="adm-segmented-item-label"
        role="option"
        title="Daily"
      >
        Daily
      </div>
    </label>
    <label
      class="adm-segmented-item adm-segmented-item-disabled"
    >
      <input
        class="adm-segmented-item-input"
        disabled=""
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title="Weekly"
      >
        Weekly
      </div>
    </label>
    <label
      class="adm-segmented-item"
    >
      <input
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title="Monthly"
      >
        Monthly
      </div>
    </label>
  </div>
</div>
`;

exports[`adm-segmented render segmented with string options 1`] = `
<div
  aria-label="segmented control"
  class="adm-segmented"
  role="listbox"
>
  <div
    class="adm-segmented-group"
  >
    <label
      class="adm-segmented-item adm-segmented-item-selected"
    >
      <input
        checked=""
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="true"
        class="adm-segmented-item-label"
        role="option"
        title="Daily"
      >
        Daily
      </div>
    </label>
    <label
      class="adm-segmented-item"
    >
      <input
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title="Weekly"
      >
        Weekly
      </div>
    </label>
    <label
      class="adm-segmented-item"
    >
      <input
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title="Monthly"
      >
        Monthly
      </div>
    </label>
  </div>
</div>
`;

exports[`adm-segmented render segmented with thumb 1`] = `
<div
  aria-label="segmented control"
  class="adm-segmented"
  role="listbox"
>
  <div
    class="adm-segmented-group"
  >
    <label
      class="adm-segmented-item adm-segmented-item-selected"
    >
      <input
        checked=""
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="true"
        class="adm-segmented-item-label"
        role="option"
        title="Map"
      >
        Map
      </div>
    </label>
    <label
      class="adm-segmented-item"
    >
      <input
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title="Transit"
      >
        Transit
      </div>
    </label>
    <label
      class="adm-segmented-item"
    >
      <input
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title="Satellite"
      >
        Satellite
      </div>
    </label>
  </div>
</div>
`;

exports[`adm-segmented render segmented: disabled 1`] = `
<div
  aria-label="segmented control"
  class="adm-segmented adm-segmented-disabled"
  role="listbox"
>
  <div
    class="adm-segmented-group"
  >
    <label
      class="adm-segmented-item adm-segmented-item-selected adm-segmented-item-disabled"
    >
      <input
        checked=""
        class="adm-segmented-item-input"
        disabled=""
        type="radio"
      />
      <div
        aria-selected="true"
        class="adm-segmented-item-label"
        role="option"
        title="Daily"
      >
        Daily
      </div>
    </label>
    <label
      class="adm-segmented-item adm-segmented-item-disabled"
    >
      <input
        class="adm-segmented-item-input"
        disabled=""
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title="Weekly"
      >
        Weekly
      </div>
    </label>
    <label
      class="adm-segmented-item adm-segmented-item-disabled"
    >
      <input
        class="adm-segmented-item-input"
        disabled=""
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
        title="Monthly"
      >
        Monthly
      </div>
    </label>
  </div>
</div>
`;

exports[`adm-segmented render with icons 1`] = `
<div
  aria-label="segmented control"
  class="adm-segmented"
  role="listbox"
>
  <div
    class="adm-segmented-group"
  >
    <label
      class="adm-segmented-item adm-segmented-item-selected"
    >
      <input
        checked=""
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="true"
        class="adm-segmented-item-label"
        role="option"
      >
        <span
          class="adm-segmented-item-icon"
        >
          <svg
            class="antd-mobile-icon"
            height="1em"
            style="vertical-align: -0.125em;"
            viewBox="0 0 48 48"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <g
              fill="none"
              fill-rule="evenodd"
              id="UnorderedListOutline-UnorderedListOutline"
              stroke="none"
              stroke-width="1"
            >
              <g
                id="UnorderedListOutline-编组"
              >
                <rect
                  fill="#FFFFFF"
                  height="48"
                  id="UnorderedListOutline-矩形"
                  opacity="0"
                  width="48"
                  x="0"
                  y="0"
                />
                <path
                  d="M8.5,32 C10.9852814,32 13,34.0147186 13,36.5 C13,38.9852814 10.9852814,41 8.5,41 C6.01471863,41 4,38.9852814 4,36.5 C4,34.0147186 6.01471863,32 8.5,32 Z M8.5,35 C7.67157288,35 7,35.6715729 7,36.5 C7,37.3284271 7.67157288,38 8.5,38 C9.32842712,38 10,37.3284271 10,36.5 C10,35.6715729 9.32842712,35 8.5,35 Z M43.6,35 C43.8209139,35 44,35.1790861 44,35.4 L44,37.6 C44,37.8209139 43.8209139,38 43.6,38 L16.4,38 C16.1790861,38 16,37.8209139 16,37.6 L16,35.4 C16,35.1790861 16.1790861,35 16.4,35 L43.6,35 Z M8.5,19 C10.9852814,19 13,21.0147186 13,23.5 C13,25.9852814 10.9852814,28 8.5,28 C6.01471863,28 4,25.9852814 4,23.5 C4,21.0147186 6.01471863,19 8.5,19 Z M8.5,22 C7.67157288,22 7,22.6715729 7,23.5 C7,24.3284271 7.67157288,25 8.5,25 C9.32842712,25 10,24.3284271 10,23.5 C10,22.6715729 9.32842712,22 8.5,22 Z M43.6,22 C43.8209139,22 44,22.1790861 44,22.4 L44,24.6 C44,24.8209139 43.8209139,25 43.6,25 L16.4,25 C16.1790861,25 16,24.8209139 16,24.6 L16,22.4 C16,22.1790861 16.1790861,22 16.4,22 L43.6,22 Z M8.5,6 C10.9852814,6 13,8.01471863 13,10.5 C13,12.9852814 10.9852814,15 8.5,15 C6.01471863,15 4,12.9852814 4,10.5 C4,8.01471863 6.01471863,6 8.5,6 Z M8.5,9 C7.67157288,9 7,9.67157288 7,10.5 C7,11.3284271 7.67157288,12 8.5,12 C9.32842712,12 10,11.3284271 10,10.5 C10,9.67157288 9.32842712,9 8.5,9 Z M43.6,9 C43.8209139,9 44,9.1790861 44,9.4 L44,11.6 C44,11.8209139 43.8209139,12 43.6,12 L16.4,12 C16.1790861,12 16,11.8209139 16,11.6 L16,9.4 C16,9.1790861 16.1790861,9 16.4,9 L43.6,9 Z"
                  fill="currentColor"
                  fill-rule="nonzero"
                  id="UnorderedListOutline-形状结合"
                />
              </g>
            </g>
          </svg>
        </span>
      </div>
    </label>
    <label
      class="adm-segmented-item"
    >
      <input
        class="adm-segmented-item-input"
        type="radio"
      />
      <div
        aria-selected="false"
        class="adm-segmented-item-label"
        role="option"
      >
        <span
          class="adm-segmented-item-icon"
        >
          <svg
            class="antd-mobile-icon"
            height="1em"
            style="vertical-align: -0.125em;"
            viewBox="0 0 48 48"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <g
              fill="none"
              fill-rule="evenodd"
              id="AppOutline-AppOutline"
              stroke="none"
              stroke-width="1"
            >
              <g
                id="AppOutline-编组"
              >
                <rect
                  fill="#FFFFFF"
                  height="48"
                  id="AppOutline-矩形"
                  opacity="0"
                  width="48"
                  x="0"
                  y="0"
                />
                <path
                  d="M38.6673097,11.5277497 L27.3505965,4.90817356 C25.2828181,3.69727548 22.7306145,3.69727548 20.6583507,4.90817356 L9.33715206,11.5277497 C7.2693737,12.7386478 6,14.9675973 6,17.3804238 L6,30.6195762 C6,33.0324027 7.27385912,35.265837 9.33715206,36.4722503 L20.6538653,43.0918264 C22.7261291,44.3027245 25.2738473,44.3027245 27.3461111,43.0918264 L38.6628243,36.4722503 C40.7306027,35.265837 41.9999882,33.0324027 41.9999882,30.6195762 L41.9999882,17.3849086 C42.0044618,14.9675973 40.7350881,12.7386478 38.6673097,11.5277497 Z M8.62397037,17.3804238 C8.62397037,15.8914677 9.40443335,14.5191165 10.673807,13.7746384 L21.9591223,7.16851665 C23.2240106,6.42852338 24.7804512,6.42852338 26.0453394,7.16851665 L32.4908872,10.9447247 L12.9479147,22.3764996 L8.62397037,19.8470681 L8.62397037,17.3804238 Z M23.7398339,19.0622267 L32.4729455,24.1704227 L24.2466863,28.9826214 L15.5135746,23.8744254 L23.7398339,19.0622267 L23.7398339,19.0622267 Z M10.673807,34.1984527 C9.40443335,33.4584595 8.62397037,32.0816235 8.62397037,30.5926673 L8.62397037,22.8474044 L21.6810263,30.485032 L13.0017397,35.5618343 L10.673807,34.1984527 L10.673807,34.1984527 Z M39.3804914,30.5926673 C39.3804914,32.0816235 38.6000284,33.4539747 37.3306547,34.1984527 L26.0453394,40.8045745 C24.7804512,41.5445678 23.2240106,41.5445678 21.9591223,40.8045745 L15.5629142,37.0597601 L35.0386054,25.6683485 L39.3804914,28.2067496 L39.3804914,30.5926673 L39.3804914,30.5926673 Z M39.3804914,25.2108981 L35.6486224,23.0267967 L26.3054938,17.5643009 L35.0565471,12.4471353 L37.3306547,13.7791232 C38.6000284,14.5191165 39.3804914,15.8959525 39.3804914,17.3849086 L39.3804914,25.2108981 L39.3804914,25.2108981 Z"
                  fill="currentColor"
                  fill-rule="nonzero"
                  id="AppOutline-形状"
                />
              </g>
            </g>
          </svg>
        </span>
        <span>
          KanbanYes
        </span>
      </div>
    </label>
  </div>
</div>
`;
